metalearners/rlearner.py

```diff
@@ -504,3 +509,30 @@ def _pseudo_outcome_and_weights(
         weights = np.square(w_residuals)
 
         return pseudo_outcomes, weights
+
+    def build_onnx(self, models: Mapping[str, Sequence], output_name: str = "tau"):
+        check_onnx_installed()
+        check_spox_installed()
+        import spox.opset.ai.onnx.v21 as op
+        from onnx.checker import check_model
+        from spox import Var, build, inline
+
+        self._validate_feature_set_all()
+        self._validate_onnx_models(models, {TREATMENT_MODEL})
+
+        input_dict = infer_input_dict(models[TREATMENT_MODEL][0])
+
+        treatment_output_name = models[TREATMENT_MODEL][0].graph.output[0].name
+
+        tau_hat: list[Var] = []
+        for m in models[TREATMENT_MODEL]:
+            tau_hat_tv = inline(m)(**input_dict)[treatment_output_name]
+            tau_hat_tv = op.unsqueeze(tau_hat_tv, axes=op.constant(value_int=2))
+            if self.is_classification:
+                tau_hat_tv = op.concat([op.neg(tau_hat_tv), tau_hat_tv], axis=-1)
```

Comment:
Iiuc this is based on the fact that the R-Learner currently doesn't support multi-class classifiction. Could we make this more explicit here, e.g. by either raising if `_supports_multi_class()` or by investigating the model to be transformed?

It'd be great if the hypothetical person who makes the R-Learner multi-class classification possible doesn't need to remember this detail but rather is told about it preemptively.

Reply:
Good catch!
[ac970e8](https://github.com/Quantco/metalearners/pull/39/commits/ac970e84aacc60365982250971b4b862cc6e1334)